lib/set.rb

```diff
@@ -168,6 +168,29 @@ def clear
     self
   end
 
+  # Removes all nil elements from self. Returns self if any elements removed,
+  # otherwise nil.
+  #
+  #     set = Set[1, 'c', nil]            #=> #<Set: {1, "c", nil}>
+  #     set.compact!                      #=> #<Set: {1, "c"}>
+  #     set                               #=> #<Set: {1, "c"}>
+  def compact!
+    original_size = @hash.size
+    @hash.delete_if { |key, _value| key.nil? }
+    original_size != @hash.size ? self : nil
```

Comment:
Why not `@hash.compact! && self`?

Reply:
Cool! I didn't come up with this idea. I updated it. Thank you!